### PR TITLE
#419 - dynamic runtime wiring factory support

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

**Comment by bbakerman** [comment]
This could go into 4.x but its an extension of the IDL work that comes in 3.x

**Comment by andimarek** [comment]
see #419

**Comment by andimarek** [comment]
@bbakerman Interesting discussion about directives in the IDL and for example how to make directives accessible: https://github.com/graphql/graphql-js/pull/746#issuecomment-301554231

**Comment by bbakerman** [comment]
Interesting that Byron as proposed what we have done (nearly once we get schema gen to transfer the node def.  which I meant to do but didnt)

> ...A simpler and more general solution would be to refer to the entire AST node associated with each definition when built with buildASTSchema. That would still allow access to directives for tooling (type.astNode.directives) but would also allow access to the rest of the AST

File: src/main/java/graphql/Assert.java [hunk 1]
```diff
@@ -10,6 +10,10 @@
         throw new AssertException(errorMessage);
     }
 
+    public static void assertNeverCalled() {
```
Comment (bbakerman) [inline src/main/java/graphql/Assert.java]: if you make this 
    public static <T> T assertNeverCalled()

then you can do this at call site

   {
     return assertShouldNeverBecalled()
   }

**Comment by bbakerman** [comment]
More than happy with the removal of the "smooshed" definition

File: src/main/java/graphql/Assert.java (hunk 1 above)
Comment (andimarek) [inline src/main/java/graphql/Assert.java]: right ... will change it

**Pull request merged**